native-image-workshop/README.md

```diff
@@ -1,51 +1,59 @@
+This workshop relies on using GraalVM 21.3 or higher, with the GraalVM Native Image package installed. It will work with either Community Edition (CE), or Enterprise Edition (EE). If there is any EE speific code / configuration, this will be clearly marked out.
 
-## Outline of Demo
+Let's beign!
 
-We are going to use a fairly trivial application to walk through some of the features that
-are available with GraalVM's Native Image. These are:
+## Setup
 
-1. How you can turn a Java app into a native executable
-2. How you can deal with Java Reflection API calls, etc..
-3. Using the command line tools for generating a native image, as well as using the Maven tooling
+Install GraalVM. Instructions can be found [here](https://www.graalvm.org/docs/getting-started/.
 
-The application is a Java command line app that counts the number of files within the current directory and sub directories.
-It also calculates their total size.
+You will need the following installed:
 
-You will need to update the code, in the following various steps, in order to work through the points we listed above.
+* GraalVM 21.3.0 JDK 11 or JDK17 installed (either EE or CE)
+* Native Image plugin that matches your GraalVM installation
 
-Let's beign!
+## Quick Note on Using Maven Profiles
 
-## Quick Note on Using Maven Profile
+This workshop will use Maven profiles. Proflies are a great way to have different build configurtions within a single `pom.xml` file. You can find out more about them [here](https://maven.apache.org/guides/introduction/introduction-to-profiles.html).
 
-The Maven build has been split into several different profiles, each of which serves a different purpose. We can call these profiles by passing a parameter that contains the name of the profile to maven. In the example below the `JAVA` profile is called:
+We will use several profiles throughout the project, each of which will serve a different purpose. We are able to call these profiles by passing a parameter containing the name of the profile to maven. The example below shows how we would call the `JAVA` profile, when building with maven:
 
 ![User Input](./images/userinput.png)
 ```sh
 $ mvn clean package -PJAVA
 ```
 
-The name of the profile to be called is appended to the `-P` flag. We have the following profiles defined in the maven file:
+The name of the profile to be called is appended to the `-P` flag. We use these profiles within this workshop (all defined in the `pom.xml` file):
+
+1. `java_agent` : Ths builds the Java application with the tracing agent
+2. `native` : This builds the native image
+
+## The Java App
 
-1. `JAVA` : This builds the Java applicaiton
-2. `JAVA_AGENT_LIB` : Ths builds the Java application with the tracing agent
-3. `NATIVE_IMAGE` : This builds the native image
+The Java application can be found in the `src` directory and it consists of two files. 
+
+* `App.java` : A wrapper for  the `ListDir` application
```

Comment:
Remove the extra space: 
A wrapper for the `ListDir` application

Reply:
updated